docs/metrics/README.md

```diff
@@ -0,0 +1,71 @@
+# Documentation
```

Comment:
can we please put this under `pkg/METRICS.md` for now please? the docs pkg is supposed to mostly hold proposals not really documentation. All documentation will eventually make it to https://github.com/kubernetes-sigs/gateway-api-inference-extension/tree/main/site-src

Reply:
Done.